examples/examples/instrumented_error.rs

```diff
@@ -0,0 +1,86 @@
+//! This example demonstrates using the `tracing-error` crate's `SpanTrace` type
+//! to attach a trace context to a custom error type.
+#![deny(rust_2018_idioms)]
+use std::error::Error;
+use std::fmt;
+use tracing_error::{prelude::*, ErrorLayer};
+use tracing_subscriber::{fmt::Layer as FmtLayer, prelude::*, registry::Registry};
+
+#[derive(Debug)]
+struct FooError {
```

Comment:
maybe a brief comment noting that this is essentially a stand-in for any type implementing `Error` (std errors, errors returned by libraries, user-defined error types, etc)?